subsys/fs/fs.c

```diff
@@ -168,6 +168,25 @@ int fs_open(struct fs_file_t *zfp, const char *file_name, fs_mode_t flags)
 		return rc;
 	}
 
+	if (flags & FS_O_TRUNC) {
+		if ((flags & FS_O_WRITE) == 0) {
```

Comment:
This check should happen before open is even attempted. Yo do not perform any operation if it is possible to predict failure.